contracts/strategies/traderjoe/TraderJoeInvestmentLib.sol

```diff
@@ -0,0 +1,319 @@
+            .pairDepositToken
+            .balanceOf(address(this));
+
+        strategyStorage.lbPair.collectFees(
+            address(this),
+            strategyStorage.binIds
+        );
+
+        uint256 pairDepositTokenAfter = strategyStorage
+            .pairDepositToken
+            .balanceOf(address(this));
+
+        uint256 pairDepositTokenIncrement = pairDepositTokenAfter -
+            pairdepositTokenBefore;
+
+        swapTokens(
+            pairDepositTokenIncrement,
+            strategyStorage.pairDepositToken,
+            strategyStorage.depositToken
+        );
+    }
+
+    function adjustBins(
+        uint256[] calldata binIds,
+        uint256[] calldata binAllocations,
+        uint256 investmentTokenSupply
+    ) public {
+        TraderJoeStorage storage strategyStorage = TraderJoeStorageLib
+            .getStorage();
+
+        // Withdraw from all bins.
+        uint256 depositTokenBefore = strategyStorage.depositToken.balanceOf(
+            address(this)
+        );
+        uint256 pairDepositTokenBefore = strategyStorage
+            .pairDepositToken
+            .balanceOf(address(this));
+
+        withdraw(investmentTokenSupply, investmentTokenSupply);
+
+        uint256 depositTokenAfter = strategyStorage.depositToken.balanceOf(
+            address(this)
+        );
+        uint256 pairDepositTokenAfter = strategyStorage
+            .pairDepositToken
+            .balanceOf(address(this));
+
+        // Set bin IDs and allocations to the given ones.
+        strategyStorage.binIds = binIds;
+        strategyStorage.binAllocations = binAllocations;
+
+        // Deposit into the new bins with the new allocations.
+        uint256 depositTokenIncrement = depositTokenAfter - depositTokenBefore;
+        uint256 pairDepositTokenIncrement = pairDepositTokenAfter -
+            pairDepositTokenBefore;
+
+        deposit(depositTokenIncrement, pairDepositTokenIncrement);
+    }
+
+    function swapTokens(
+        uint256 amountIn,
+        IERC20Upgradeable tokenIn,
+        IERC20Upgradeable tokenOut
+    ) public returns (uint256 amountOut) {
+        TraderJoeStorage storage strategyStorage = TraderJoeStorageLib
+            .getStorage();
+
+        (amountOut, ) = strategyStorage.lbRouter.getSwapOut(
+            address(strategyStorage.lbPair),
+            amountIn,
+            strategyStorage.tokenY == strategyStorage.depositToken
+        );
+
+        if (amountOut == 0) {
+            return 0;
+        }
+
+        address[] memory path = new address[](2);
+        path[0] = address(tokenIn);
+        path[1] = address(tokenOut);
+
+        uint256[] memory binSteps = new uint256[](1);
+        binSteps[0] = strategyStorage.binStep;
+
+        amountOut = SwapServiceLib.swapExactTokensForTokens(
+            strategyStorage.swapService,
+            amountIn,
+            0,
+            path,
+            binSteps
+        );
+    }
+
+    function swapBack(uint256 amountIn) public {
```

Comment:
if I just see '__swapBack' I have no idea what it means, I need to look into the implementation to undersand it. Can we call this __swapBackToDepositCurrency?

Reply:
I'm not that fan of comments, but since you're asking, I added one above it. Does it still confuse you?

Reply:
I didn't mean comments, I agree that in thsi case comments are not a good idea. However whenever someone looks at a function name (or variable name), it should be self-explanatory what the function does. We should not look into the implementation to know what is swapped back to what. 

Sorry if you feel I am picking on naming convention, but at least the function names should be self-descreptive.